File: matlab_examples/InputChannelCalibration.m
clc;
clear all;

% EA Engine installation path
installPath = 'C:\Program Files\Bruel and Kjaer\EA Engine';
% EA Engine executable filename (including full path)
assemblyName = 'C:\Program Files\Bruel and Kjaer\EA Engine\EA_Engine.exe';

% Add the EA Engine .NET assembly (please note that it cannot be removed from the current MATLAB session until exiting)
EA_Engine = NET.addAssembly(assemblyName);
% Import the assembly namespace
import EA_Engine.*;

% Declare and instanciate a new Engine object (the installation path is mandatory)
engine = Engine(installPath);

% Listeners for a .NET object (here the engine object)
% Events are:
% - Feedback: returns information during a task execution
% - CalibrationResultsUpdated: returns the updated results during the measurement (concerns only input or output callibration, the other tests don't raise this event)
% - TestResultsAvailable: returns the test results once the acquisition and processing are finished (concerns all the tasks)
lh_Feedback = addlistener(engine, 'Feedback', @Feedback);
lh_CalibrationResultsUpdated = addlistener(engine,'CalibrationResultsUpdated',@CalibrationResultsUpdated);
%lh_CalibrationResultsAvailable = addlistener(engine, 'TestResultsAvailable', @CalibrationResultsAvailable);

% Set the input calibration settings
calibrationSettings = engine.SetInputCalibrationSettings(1000, 94, 'dB', 5, ResultFileFormatTypes.XML, true, '2,3', 8192);

% Define the arguments for the first input channel calibration (see user's manual for details)
arguments = sprintf('%s %s', 'Calibrate_Input_Channel', '1');
% Execute the task
engine.Execute(arguments);

% Delete the listeners once the task is finished
delete (lh_Feedback);
delete (lh_CalibrationResultsUpdated);
%delete (lh_CalibrationResultsAvailable);

File: python_examples/Input Channel Calibration.py
import pythoncom
import clr

pythoncom.CoInitialize()

# EA Engine installation path
installPath = "C:\\Program Files\\Bruel and Kjaer\\EA Engine"
# Add the EA Engine .NET assembly (EA Engine executable filename - including full path)
clr.AddReference("C:\\Program Files\\Bruel and Kjaer\\EA Engine\\EA_Engine.exe")
# Import the assembly namespace
from EA_Engine import *

# Import the functions used by handlers
from Feedback import feedback
from CalibrationResultsUpdated import calibrationresultsupdated
from CalibrationResultsAvailable import calibrationresultsavailable

# Declare and instanciate a new Engine object (the installation path is mandatory)
engine = Engine(installPath)

# Add handlers for a .NET object (here the engine object)
# Events are:
# - Feedback: returns information during a task execution
# - CalibrationResultsUpdated: returns the updated results during the measurement (concerns only input or output callibration, the other tests don't raise this event)
# - TestResultsAvailable: returns the test results once the acquisition and processing are finished (concerns all the tasks)
engine.Feedback += feedback
engine.CalibrationResultsUpdated += calibrationresultsupdated
engine.TestResultsAvailable += calibrationresultsavailable

# Set the input calibration settings
calibrationSettings = CalibrationSettings()
calibrationSettings.ReferenceFrequency = 1000
calibrationSettings.ReferenceLevel = 94
calibrationSettings.ReferenceUnit = "dB"
calibrationSettings.Duration = 10
calibrationSettings.ResultFileFormatType = ResultFileFormatTypes.CSV
calibrationSettings.CalculateTHD = True
calibrationSettings.HarmonicsTHD = "2,3"
calibrationSettings.AnalysisFFTSettings.BlockSize = 8192

calibrationSettings = engine.SetInputCalibrationSettings(calibrationSettings)

# Execute the task (calibrate input channel 1)
engine.Execute("Calibrate_Input_Channel 1")

# Remove handlers
engine.Feedback -= feedback
engine.CalibrationResultsUpdated -= calibrationresultsupdated
engine.TestResultsAvailable -= calibrationresultsavailable
